lib/node_modules/@stdlib/math/base/complex/divide/test/test.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var cdiv = require( './../lib' );
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof cdiv, 'function', 'main export is a function' );
+	t.end();
+});
+
+function testCdiv( a, b, c, d, e, f, label ) {
+	tape(label, function test( t ) {
+		var expected;
+		var v;
+
+		v = cdiv( a, b, c, d );
+		expected = [ e, f ];
+
+		t.deepEqual( v, expected, '( ' + a + ' + ' + b + 'j ) / ( ' + c + ' + ' + d + 'j ) == ' + e + ' + ' + f + 'j' );
+
+		t.end();
+	});
+}
+
+testCdiv( -13.0, -1.0, -2.0, 1.0, 5.0, 3.0, 'simple division problem' );
+testCdiv( 1e308, 5e307, 1.0, 0.5, 1e308, 0, 'ab >= LARGE_THRESHOLD' );
+testCdiv( 1, 0.5, 1e308, 5e307, 1e-308, 0, 'cd >= LARGE_THRESHOLD' );
+testCdiv( 1e-304, 2e-304, 1, 2, 1e-304, 0, 'ab <= SMALL_THRESHOLD');
+testCdiv( 1, 2, 1e-304, 2e-304, 1e+304, 0, 'cd <= SMALL_THRESHOLD');
+testCdiv( 2, 4, 0, 2, 2, -1, 'r === 0 branch of internalCompreal' );
```

Comment:
We try to avoid "leaking" implementation details in our tests. Meaning, a user reading the tests as user documentation should never need to know about a `r === 0` branch. Instead, we try to reframe descriptions in terms of special combinations of input arguments, or higher-level concepts.